### PR TITLE
Add a changelog entry for #2018

### PR DESCRIPTION
**Pull request by sammy-tri**
As suggested by @jwnimmer-tri in #2018.

### CODE REVIEW

**Comment by liangfok** [comment]
May need to run pimpmychangelog.

**Comment by ggould-tri** [comment]
Reviewed:  Looks good to me.

**Comment by jwnimmer-tri** [comment]
@sammy-tri Yes, you need to add the 2018 link at the bottom (just do it manually, don't need the pimp tool.)

**Comment by sammy-tri** [comment]
Added link

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,7 +24,8 @@ Unreleased: changes on master, not yet released
  - [#1992][] Matlab tests must now be explicitly listed in CMakeLists.
  - [#1970][] The `drake` pod Makefile now includes the `install` action.
  - [#1953][] Replace `valuecheckMatrix()` with `CompareMatrices()`.
- 
+ - [#2018][] Fix capitalization of Constrant and OptimizationProblem APIs to match style guide.
```
Comment (jwnimmer-tri) [inline CHANGELOG.md]: So far in the life of this changelog, we have been fairly pedantic in backticking names that are code.  I am okay if we want to stop doing that, but we should be consistent.

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,7 +24,8 @@ Unreleased: changes on master, not yet released
  - [#1992][] Matlab tests must now be explicitly listed in CMakeLists.
  - [#1970][] The `drake` pod Makefile now includes the `install` action.
  - [#1953][] Replace `valuecheckMatrix()` with `CompareMatrices()`.
- 
+ - [#2018][] Fix capitalization of `Constrant` and `OptimizationProblem` APIs to match style guide.
```
Comment (Lucy-tri) [inline CHANGELOG.md]: Did you mean Constraint?

**Comment by david-german-tri** [comment]
Jenkins failure is just a Windows upload hiccup.  Merging.

**Comment by david-german-tri** [comment]
Opened #2035.

**Pull request merged**